frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/GitCommandUpdatingCurrentBranchBackgroundable.java

```diff
@@ -0,0 +1,206 @@
+
+    GitUpdatedRanges updatedRanges = deriveGitUpdatedRanges(getTargetBranchName());
+
+    String beforeRevision = gitRepository.getCurrentRevision();
+    try (AccessToken ignore = DvcsUtil.workingTreeChangeStarted(project, getOperationName())) {
+      GitCommandResult result = Git.getInstance().runCommand(handler);
+
+      if (beforeRevision != null) {
+        GitRevisionNumber currentRev = new GitRevisionNumber(beforeRevision);
+        handleResult(result,
+            localChangesDetector,
+            untrackedFilesDetector,
+            currentRev,
+            beforeLabel,
+            updatedRanges);
+      }
+    }
+  }
+
+  private @Nullable GitUpdatedRanges deriveGitUpdatedRanges(String remoteBranchName) {
+    GitUpdatedRanges updatedRanges = null;
+    var currentBranch = gitRepository.getCurrentBranch();
+    if (currentBranch != null) {
+      GitBranch targetBranch = gitRepository.getBranches().findBranchByName(remoteBranchName);
+      if (targetBranch != null) {
+        GitBranchPair refPair = new GitBranchPair(currentBranch, targetBranch);
+        updatedRanges = GitUpdatedRanges.calcInitialPositions(project,
+            java.util.Collections.singletonMap(gitRepository, refPair));
+      } else {
+        LOG.warn("Couldn't find the branch with name '${remoteBranchName}'");
+      }
+    }
+    return updatedRanges;
+  }
+
+  private void handleResult(
+      GitCommandResult result,
+      GitLocalChangesWouldBeOverwrittenDetector localChangesDetector,
+      GitUntrackedFilesOverwrittenByOperationDetector untrackedFilesDetector,
+      GitRevisionNumber currentRev,
+      Label beforeLabel,
+      @Nullable GitUpdatedRanges updatedRanges) {
+    VirtualFile root = gitRepository.getRoot();
+    if (result.success()) {
+      VfsUtil.markDirtyAndRefresh(/* async */ false, /* recursive */ true, /* reloadChildren */ false, root);
+      gitRepository.update();
+      if (updatedRanges != null &&
+          AbstractCommonUpdateAction
+              .showsCustomNotification(java.util.Collections.singletonList(GitVcs.getInstance(project)))) {
+        var ranges = updatedRanges.calcCurrentPositions();
+        GitUpdateInfoAsLog.NotificationData notificationData = new GitUpdateInfoAsLog(project, ranges)
+            .calculateDataAndCreateLogTab();
+
+        Notification notification;
+        if (notificationData != null) {
+          String title = getTitleForUpdateNotification(notificationData.getUpdatedFilesCount(),
+              notificationData.getReceivedCommitsCount());
+          String content = getBodyForUpdateNotification(notificationData.getFilteredCommitsCount());
+          notification = VcsNotifier.STANDARD_NOTIFICATION.createNotification(title,
+              content,
+              INFORMATION,
+              /* listener */ null);
+          notification.addAction(NotificationAction.createSimple(getString(
```

Comment:
Why not just `VcsNotifier.getInstance(project).notifyInfo(...)` here?

Reply:
Hmm good question... @mkondratek :question:

Reply:
I took a cue from `git4idea.actions.GitMergeAction#handleResult`

@micpiotrowski how would you provide the `View commits` action to the notification using `VcsNotifier.getInstance(project).notifyInfo(...)`?

Reply:
Yeah, I didn't notice that. My fault.